packages/mobile/src/verify/reducer.ts

```diff
@@ -0,0 +1,366 @@
+  'VERIFY/SET_VERIFICATION_STATUS'
+)
+export const setActionableAttestation = createAction<ActionableAttestation[]>(
+  'VERIFY/SET_ACTIONABLE_ATTESTATIONS'
+)
+
+export enum StateType {
+  Idle = 'Idle',
+  Preparing = 'Preparing',
+  EnsuringRealHumanUser = 'EnsuringRealHumanUser',
+  StartingKomenciSession = 'StartingKomenciSession',
+  FetchingPhoneNumberDetails = 'FetchingPhoneNumberDetails',
+  FetchingMtw = 'FetchingMtw',
+  FetchingOnChainData = 'FetchingOnChainData',
+  Error = 'Error',
+}
+
+// Idle State
+interface Idle {
+  type: StateType.Idle
+}
+export const idle = (): Idle => ({ type: StateType.Idle })
+
+// PreparingKomenci State
+interface Preparing {
+  type: StateType.Preparing
+}
+// {}: Omit<PreparingKomenci, 'type'>
+export const preparing = (): Preparing => ({
+  type: StateType.Preparing,
+})
+
+// EnsuringRealHumanUser State
+interface EnsuringRealHumanUser {
+  type: StateType.EnsuringRealHumanUser
+}
+export const ensuringRealHumanUser = (): EnsuringRealHumanUser => ({
+  type: StateType.EnsuringRealHumanUser,
+})
+
+// StartingKomenciSession State
+interface StartingKomenciSession {
+  type: StateType.StartingKomenciSession
+}
+export const startingKomenciSession = (): StartingKomenciSession => ({
+  type: StateType.StartingKomenciSession,
+})
+
+// FetchingPhoneNumberDetails State
+interface FetchingPhoneNumberDetails {
+  type: StateType.FetchingPhoneNumberDetails
+}
+export const fetchingPhoneNumberDetails = (): FetchingPhoneNumberDetails => ({
+  type: StateType.FetchingPhoneNumberDetails,
+})
+
+// FetchingMtw State
+interface FetchingMtw {
+  type: StateType.FetchingMtw
+}
+export const fetchingMtw = (): FetchingMtw => ({
+  type: StateType.FetchingMtw,
+})
+
+// FetchingVerificationOnChain State
+interface FetchingOnChainData {
+  type: StateType.FetchingOnChainData
+}
+export const fetchingOnChainData = (): FetchingOnChainData => ({
+  type: StateType.FetchingOnChainData,
+})
+
+// Error State
+interface Error {
+  type: StateType.Error
+  message: string
+}
+export const error = (message: string): Error => ({
+  type: StateType.Error,
+  message,
+})
+
+type InternalState =
+  | Idle
+  | Preparing
+  | EnsuringRealHumanUser
+  | StartingKomenciSession
+  | FetchingPhoneNumberDetails
+  | FetchingMtw
+  | FetchingOnChainData
+  | Error
+
+export interface KomenciContext {
+  errorTimestamps: number[]
+  unverifiedMtwAddress: string | null
+  sessionActive: boolean
+  sessionToken: string
+  callbackUrl: string | undefined
+  captchaToken: string
+}
+
+export enum KomenciAvailable {
+  Yes = 'YES',
+  No = 'NO',
+  Unknown = 'UNKNOWN',
+}
+
+export interface State {
+  status: AttestationsStatus & { komenci: boolean }
+  actionableAttestations: ActionableAttestation[]
+  currentState: InternalState
+  komenci: KomenciContext
+  komenciAvailable: KomenciAvailable
+  phoneHash?: string
+  e164Number?: string
+  retries: number
+  withoutRevealing: boolean
+  TEMPORARY_override_withoutVerification?: boolean
```

Comment:
This is a dirty trick, so the functionality can work before the second PR is merged. It will be removed in the second PR: https://github.com/celo-org/wallet/pull/116/files#diff-4d5ede444b6baea8f4bd12874e556f8689a90d97396b6988752b99a321af4794